include/bitcoin/bitcoin/wallet/qrcode.hpp

```diff
@@ -29,60 +29,53 @@
 
 #ifdef WITH_QRENCODE
 #include "qrencode.h"
-#endif
-
 
 namespace libbitcoin {
 namespace wallet {
 
-#ifdef WITH_QRENCODE
-
-typedef uint32_t qr_version;
-typedef QRecLevel qr_rec_level;
-typedef QRencodeMode qr_encode_mode;
-typedef uint32_t qr_case_sensitive;
-
-namespace qrcode {
-
-BC_CONSTEXPR qr_version version = 0;
-BC_CONSTEXPR qr_rec_level level = QR_ECLEVEL_L;
-BC_CONSTEXPR qr_encode_mode hint = QR_MODE_8;
-BC_CONSTEXPR qr_case_sensitive case_sensitive = 1;
-
-} // namespace qrcode
-
-/**
- * A method that takes an input stream and writes the encoded qr data
- * to the specified output stream with default parameter values.
- */
-BC_API bool qrencode_data(std::istream& in, std::ostream& out);
-
-/**
- * A method that takes a data chunk and returns the encoded qr data as
- * a data_chunk with default parameter values.
- */
-BC_API data_chunk qrencode_data(const data_chunk& data);
-
-/**
- * A method that takes a data chunk and returns the encoded qr data as
- * a data chunk with the specified parameter values.
- */
-BC_API data_chunk qrencode_data(const data_chunk& data,
-    const qr_version version, const qr_rec_level level,
-    const qr_encode_mode hint, const qr_case_sensitive cs);
-
-/**
- * A method that takes an input stream and writes the encoded qr data
- * to the output stream with the specified parameter values.
- */
-BC_API bool qrencode_data(std::istream& in, const qr_version version,
-    const qr_rec_level level, const qr_encode_mode hint,
-    const qr_case_sensitive cs, std::ostream& out);
-
-#endif
+class BC_API qr
+{
+    typedef QRencodeMode encode_mode;
+    typedef QRecLevel error_recovery_level;
+
+  public:
+    static constexpr uint32_t version = 0;
```

Comment:
Headers need to use `BC_CONSTEXPR` vs. `constexpr`. This allows library linkage into VC++ 12.0 compiled builds (without CTP upgrade). This is a requirement for working with QT in MSVC.